src/main/java/com/github/jonatino/process/DataSource.java

```diff
@@ -27,6 +27,8 @@
 public interface DataSource {
 
 	MemoryBuffer read(Pointer address, int size);
+	
+	MemoryBuffer read(Pointer address, int size, MemoryBuffer tobuf);
```

Comment:
Since we added this I think we can change remove `MemoryBuffer read(Pointer address, int size);` and change

```
 	default MemoryBuffer read(long address, int size) {
  		return read(Cacheable.pointer(address), size, Cacheable.buffer(size));
  	}
```

otherwise we would need to override two methods which do the same thing.

Also what environment did you experience the object creation on? Was it Linux, Windows or Mac?